### ISSUE TITLE
[0.9.2] Server Error 500 following CQ timeout

### ISSUE THREAD

**Issue by mnietz** [influxdb/influxdb#3709]
InfluxDB-0.9.2
Debian-8.1

I've had a strange issue with the above configuration. In the influx-logs i recognized, that at some point of time a continuous query died with a timeout (btw: what's the maximum time for a cq ?):

```
[continuous_querier] 2015/08/15 03:00:17 timeout
[continuous_querier] 2015/08/15 03:00:17 error: timeout. running: SELECT count(distinct(sid)) AS "value" INTO "db"."policy01".view_distinct_per_sid_1h FROM "db"."default".view WHERE time >= '2015-08-15T01:00:00Z' AND time < '2015-08-15T02:00:00Z' GROUP BY time(1h), *
[continuous_querier] 2015/08/15 03:00:17 error executing query: CREATE CONTINUOUS QUERY view_distinct_per_sid_1h ON db BEGIN SELECT count(distinct(sid)) AS "value" INTO "db"."policy01".view_distinct_per_sid_1h FROM "db"."default".view GROUP BY time(1h), * END: err = timeout
```
This timeout was possibly causes by a system backup running at that time, but a few minutes after that, inserting into the database were no longer possible and result in a 500 error

```
[http] 2015/08/15 03:05:02 [15/Aug/2015:03:05:02 +0200] POST /write?db=db&p=%5BREDACTED%5D&precision=s&u=user HTTP/1.1 204 0 - -a86dcd11-42e9-11e5-a5a8-000000000000 1.512026ms
[http] 2015/08/15 03:05:02 [15/Aug/2015:03:05:02 +0200] POST /write?db=db&p=%5BREDACTED%5D&precision=s&u=user HTTP/1.1 204 0 - - a875d87b-42e9-11e5-a5a9-000000000000 1.661902ms

[http] 2015/08/15 03:05:07 [15/Aug/2015:03:05:02 +0200] POST /write?db=db&p=%5BREDACTED%5D&precision=s&u=user HTTP/1.1 500 8 - - a87f2cd2-42e9-11e5-a5aa-000000000000 5.000482862s
[http] 2015/08/15 03:05:12 [15/Aug/2015:03:05:07 +0200] POST /write?db=db&p=%5BREDACTED%5D&precision=s&u=user HTTP/1.1 500 8 - - 
ab814124-42e9-11e5-a5ab-000000000000 5.000432216s
```

There are no long-entries missing between these events. Normaly our writes last about 1-2 ms, than suddenly it is constant over 5s and returning 500 error.

We 'solved' this by restarting influx.

**Comment by jfromaniello** [comment]
I am having the same issue when writing from node.js many records.

I eventually start getting 500s in the logs, the node.js sdk gets a "timeout" error.

**Comment by mnietz** [comment]
In my case it's a simple php-script that uses curl to insert data.

**Comment by jwilder** [comment]
Can you test with the current nightly?  Continuous queries had a known issue that would cause timeouts.  It was fixed by #3517 #3522.

**Comment by jfromaniello** [comment]
I think I fixed my issue just by upgrading my EC2 instance. It seems it was running out of RAM and/or cpu. I will keep an eye on.

**Comment by mnietz** [comment]
Will try it tomorrow, but the error happens unregulary, so it will take some time to judge if it's fixed.

**Comment by beckettsean** [comment]
@mnietz any updates on reproducing with 0.9.3-rc2 or a nightly?

**Comment by beckettsean** [comment]
Believed fixed by 0.9.3 performance changes.
